projects/vdk-core/src/vdk/internal/builtin_plugins/templates/template_impl.py

```diff
@@ -45,7 +46,13 @@ def execute_template(self, name: str, template_args: Dict) -> ExecutionResult:
         template_job = self._datajob_factory.new_datajob(
             template_directory, self._core_context, name=self._job_name
         )
-        result = template_job.run(template_args)
+        self._core_context.state.set(ExecutionStateStoreKeys.TEMPLATE_RUNNING, True)
```

Comment:
Oh and yes, I think it's better idea to record teh name of the template - as it's valuable information - instead of just saying true/false

Reply:
Added the full information, about name and args; 
I noticed, when calling a template with arguments from "outside", it seems we're not keeping a contract of immutability. What happens then, in case the dictionary values are modified during template execution?

Reply:
On the note to transfer the state to the child context,
I noticed in https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/core/context.py#L48:
```
child_context.state.set(StoreKey[StateStore]("vdk.state.parent"), self.state)
children_key = ImmutableStoreKey[List[StateStore]]("vdk.state.children")
if self.state.get(children_key) is None:
    self.state.set(children_key, [])
self.state.get(children_key).append(child_context.state)
```

Seems we're design-prepared to spawn multiple child contexts ..at the same time. So, in case a `bool TEMPLATE_RUNNING True`, and "active" child context, can't we retrieve the child data job name (template name) and arguments (template args) by the context itself? Should we duplicate?

On the other topic, where the flag should be (parent or child), I still think the flag should be outside the template run process itself.